#24 pre-cache enviroment prior to verifyWithHCaptcha call (#25)
hCaptcha/hcaptcha-android-sdk: README.md

    });
```

You can also pass `YOUR_API_SITE_KEY` via `AndroidManifest.xml` like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest ...>
    ...
    <application ...>
        <meta-data
            android:name="com.hcaptcha.sdk.site-key"
            android:value="YOUR_API_SITE_KEY" />
       ...
    </application>
</manifest>
```

And then simply call:

```java
HCaptcha.getClient(this).verifyWithHCaptcha()...
```

> If you need to customize the look and feel, language, endpoint, etc. this can be achieved by passing a `HCaptchaConfig` object to `getClient` method.

Reviewer: This is outdated.

Author: Done

                .resetOnTimeout(true)
                .theme(HCaptchaTheme.DARK)
                .build();

> HCaptcha.getClient(this, config).verifyWithHCaptcha()...;

Reviewer: This would be a breaking change. 

Can we somehow maintain the old api and add a new parameter to the `HCaptchaConfig.lazy = true/false` which would basically just not `fragment.show()` but require an explicit `.show()` to be called to show the hCaptcha view. Thus, the new api is not a breaking change and allows for both old and new lazy approach.

 What do you reckon @CAMOBAP @e271828- ?

Example:

```java
final HCaptchaConfig config = HCaptchaConfig.builder()
                ...
                .theme(HCaptchaTheme.DARK)
                .lazy(true)
                .build();
final HCaptcha hCaptchaClient = HCaptcha.getClient(this).verifyWithHCaptcha(config);

// Async when ready to show, client would do:
hCaptchaClient.show()
  .addOnSuccessListener(new OnSuccessListener<HCaptchaTokenResponse>() {
        @Override
        public void onSuccess(HCaptchaTokenResponse response) {
            String userResponseToken = response.getTokenResult();
            // Validate the user response token using the hCAPTCHA siteverify API.
        }
    })
    .addOnFailureListener(new OnFailureListener() {
        @Override
        public void onFailure(HCaptchaException e) {
            Log.d("MainActivity", "Error code: " + e.getStatusCode());
            Log.d("MainActivity", "Error msg: " + e.getMessage());
        }
    });
```

Reviewer 2: This seems like a better approach, yeah.

Author: > This would be a breaking change.

Definitely

> What do you reckon @CAMOBAP @e271828- ?

In this case, API becomes less clear for me because depends on `lazy` param `verifyWithHCaptcha` will work significantly different, + extra API method `show`

So between `less clear API` vs `breaking change` I would select the first one

Author: Looks great, will update PR soon

Author: Done

HCaptcha.getClient(this).verifyWithHCaptcha(config)...;
```

Also to reduce hCaptcha start time, `setup` can be used:

```java
HCaptcha hCaptcha = HCaptcha.getClient(this).setup()
```

`setup` accepts the same arguments as `verifyWithHCaptcha`. Once `setup` is called, later you can call `verifyWithHCaptcha` with no arguments

> If `verifyWithHCaptcha` will be called with different arguments than `setup`, SDK will handle this and re-configure hCaptcha.

Reviewer 2: ```suggestion
If `verifyWithHCaptcha` is called with different arguments than `setup` the SDK will handle this by re-configuring hCaptcha.
```

Author: I would say that with a new config object it will "work as before", so no benefits, because it will create a fragment instance.

We added "load from the cache" to WebView, but it doesn't look related to this place in the documentation
